libclamav_rust/build.rs

```diff
@@ -48,6 +53,47 @@ fn main() {
             }
         }
     }
+
+    let cargo_cmd = env::var("CARGO_CMD").unwrap_or("".into());
+    // Only generate the bindings if CARGO_TARGET_DIR is defined, which it will
+    // be when cmake invokes `cargo build`.
+    if cargo_cmd == "build" {
+        let build_dir = PathBuf::from(env::var("CARGO_TARGET_DIR").unwrap_or(".".into()));
+        let cc_include_path_build = format!("-I{}", build_dir.join("..").to_str().unwrap());
+
+        // Configure and generate bindings.
+        let bindings = builder()
+            // Silence code-style warnings for generated bindings.
+            .raw_line("#![allow(non_snake_case, non_camel_case_types, non_upper_case_globals)]")
+            // Generate bindings for inline functions, like "cli_dbgmsg".
+            .generate_inline_functions(true)
+            // Make the bindings pretty.
+            .rustfmt_bindings(true)
+            // Enable bindgen to find the headers.
+            .clang_arg("-I../libclamav")
+            .clang_arg("-I../libclamunrar_iface")
+            .clang_arg("-I../libclammspack")
+            .clang_arg(cc_include_path_build)
+            // Generate bindings for *at least* the following.
+            .header("../libclamav/matcher.h")
+            .allowlist_type("cli_matcher")
+            .header("../libclamav/matcher-ac.h")
+            .allowlist_type("cli_ac_data")
+            .allowlist_type("cli_ac_result")
+            .header("../libclamav/others.h")
+            .allowlist_function("cli_ctx")
```

Comment:
Can we define these `const`, as we did with the other things we inspect (e.g., `LIB*`), and then just repeatedly update `bindings`?

Reply:
That's an idea. 👍 I'll give it a try.

Reply:
Updated.  Better?